### PR TITLE
If MAM lookup code crashes, user gets an empty result set instead of an error

### PR DESCRIPTION
**Pull request by arcusfelis**
This PR addresses MIM-2101

Proposed changes include:
* If message fails to be decoded, we return a stanza saying so.
* Each failed message would get such a stanza.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/203049) / elasticsearch_and_cassandra_mnesia / f274d4dcb963d1a6379d4858e62a0a6067ce40fb
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4191/203049/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4191/203049/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40d8070d1d1076.2023-12-12_17.30.36/index.html)
**OK: 368** / **Failed: 2** / **User-skipped: 40** / Auto-skipped: 0 
<details><summary>mam_SUITE:cassandra_muc_light:muc_light_failed_to_decode_message_in_database</summary>


```erlang
{error,{test_case_failed,"Unexpected error IQ."}}
```

[Report log](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4191/203049/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40d8070d1d1076.2023-12-12_17.30.36/big_tests.tests.mam_SUITE.logs/run.2023-12-12_17.30.38/mam_suite.muc_light_failed_to_decode_message_in_database.html)
</details>
<details><summary>mam_SUITE:cassandra_impl_specific:pm_failed_to_decode_message_in_database</summary>


```erlang
{error,{test_case_failed,"Unexpected error IQ."}}
```

[Report log](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4191/203049/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40d8070d1d1076.2023-12-12_17.30.36/big_tests.tests.mam_SUITE.logs/run.2023-12-12_17.30.38/mam_suite.pm_failed_to_decode_message_in_database.html)
</details>

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/203085) / elasticsearch_and_cassandra_mnesia / f31d3ee5e777145984636cc5b936d79fbcc73f08
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4191/203085/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4191/203085/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40cf64ba31c7f3.2023-12-12_18.26.13/index.html)
**OK: 369** / **Failed: 1** / **User-skipped: 40** / Auto-skipped: 0 
<details><summary>mam_SUITE:cassandra_muc_light:muc_light_failed_to_decode_message_in_database</summary>


```erlang
{error,{test_case_failed,"Unexpected error IQ."}}
```

[Report log](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4191/203085/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40cf64ba31c7f3.2023-12-12_18.26.13/big_tests.tests.mam_SUITE.logs/run.2023-12-12_18.26.14/mam_suite.muc_light_failed_to_decode_message_in_database.html)
</details>

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/203123) / elasticsearch_and_cassandra_mnesia / 87d01acbc290aeda34910804c1e8b77d5dbc5159
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4191/203123/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4191/203123/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40d985929b1ebe.2023-12-12_23.34.30/index.html)
**OK: 370** / Failed: 0 / **User-skipped: 40** / Auto-skipped: 0

File: big_tests/tests/mam_SUITE.erl [hunk 1]
```diff
@@ -538,7 +542,8 @@ prefs_cases() ->
      run_set_and_get_prefs_cases].
 
 impl_specific() ->
-    [check_user_exist].
+    [check_user_exist,
+     pm_failed_to_decode_message_in_database].
```
Comment (NelsonVides) [inline big_tests/tests/mam_SUITE.erl:546]: Why in this group and not in some regular pm one? Just asking 🤔

**Review by NelsonVides — APPROVED** [review]
LGTM

File: big_tests/tests/mam_SUITE.erl (hunk 1 above)
Comment (arcusfelis) [reply big_tests/tests/mam_SUITE.erl:546]: cannot be run in parallel. because it changes module config mid test.

**Pull request merged**